gfx.h

```diff
@@ -2226,6 +2230,237 @@ class GfxInternal
         return kGfxResult_NoError;
     }
 
+    GfxTexture resizeTexture2D(GfxTexture texture, uint32_t width, uint32_t height, uint32_t mip_levels)
+    {
+        if(!texture) return texture;
+        if(!texture_handles_.has_handle(texture.handle))
+        {
+            GFX_PRINT_ERROR(kGfxResult_InvalidOperation, "Cannot resize invalid texture object");
+            return texture;
+        }
+        Texture &gfx_texture = textures_[texture];
+        if((gfx_texture.flags_ & Texture::kFlag_AutoResize) != 0)
+        {
+            GFX_PRINT_ERROR(kGfxResult_InvalidOperation, "Cannot manually resize auto sized texture object");
+            return texture;
+        }
+        ID3D12Resource *resource = nullptr;
+        D3D12MA::Allocation *allocation = nullptr;
+        D3D12_RESOURCE_STATES const resource_state = D3D12_RESOURCE_STATE_COPY_DEST;
+        D3D12_RESOURCE_DESC
+        resource_desc                            = {};
+        resource_desc.Dimension                  = D3D12_RESOURCE_DIMENSION_TEXTURE2D;
+        resource_desc.Width                      = width;
+        resource_desc.Height                     = height;
+        resource_desc.DepthOrArraySize           = 1;
+        resource_desc.MipLevels                  = (uint16_t)mip_levels;
+        resource_desc.Format                     = texture.getFormat();
+        resource_desc.SampleDesc.Count           = 1;
+        D3D12MA::ALLOCATION_DESC allocation_desc = {};
+        allocation_desc.HeapType = D3D12_HEAP_TYPE_DEFAULT;
+        if(createResource(allocation_desc, resource_desc, resource_state, gfx_texture.clear_value_, &allocation, IID_PPV_ARGS(&resource)) != kGfxResult_NoError)
+        {
+            GFX_PRINT_ERROR(kGfxResult_OutOfMemory, "Unable to resize 2D texture object of size %ux%u", width, height);
+            return texture;
+        }
+        collect(gfx_texture); // release previous texture
+        uint64_t handle = texture_handles_.allocate_handle();
+        Texture &new_texture = textures_.insert(handle & 0xFFFFFFFFull);
+        memcpy(new_texture.clear_value_, gfx_texture.clear_value_, sizeof(gfx_texture.clear_value_));
+        new_texture.resource_ = resource;
+        new_texture.allocation_ = allocation;
+        new_texture.resource_state_ = resource_state;
+        textures_.erase(texture); // destroy old texture object
+        texture_handles_.free_handle(texture.handle);
+        texture.handle = handle;
```

Comment:
something somewhere seems to hold and use the handle, If reusing the same handle then you get rendering errors (even when the texture objects are identical) so this creates a new handle